public/app/features/dashboard/components/GenAI/QuickFeedback.tsx

```diff
@@ -0,0 +1,57 @@
+import { css } from '@emotion/css';
+import React from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Button, useStyles2 } from '@grafana/ui';
+
+import { QuickFeedbackType } from './utils';
+
+interface QuickActionsProps {
+  onSuggestionClick: (suggestion: QuickFeedbackType) => void;
+  isGenerating: boolean;
+}
+
+export const QuickFeedback = ({ onSuggestionClick, isGenerating }: QuickActionsProps) => {
+  const styles = useStyles2(getStyles);
+
+  return (
+    <div className={styles.quickSuggestionsWrapper}>
+      <Button
+        onClick={() => onSuggestionClick(QuickFeedbackType.Shorter)}
+        size="sm"
+        variant="secondary"
+        disabled={isGenerating}
+      >
+        {QuickFeedbackType.Shorter}
+      </Button>
+      <Button
+        onClick={() => onSuggestionClick(QuickFeedbackType.MoreDescriptive)}
+        size="sm"
+        variant="secondary"
+        disabled={isGenerating}
+      >
+        {QuickFeedbackType.MoreDescriptive}
+      </Button>
+      <Button
+        onClick={() => onSuggestionClick(QuickFeedbackType.Regenerate)}
+        size="sm"
+        variant="secondary"
+        disabled={isGenerating}
+      >
+        {QuickFeedbackType.Regenerate}
```

Comment:
IMO it may make sense to only append a new entry to history when actual written input is provided by user and when using the "regenerate" option is called. The other quick feedback options could just modify the current text in place. In the current form I feel that we might be adding too many entries to history that are small adjustments 🤷‍♂️

Reply:
Not sure about this... it could be confusing for the user to get sometimes a new entry and some other times not. It is not clear from the UX they are different. I'd try to go for the simplest approach and get the entries always there

Reply:
Agreed - as discussed in today's sync I think we should keep the current behavior of adding a new history item as a result of any action 👍